src/enigma2/data/Channel.h

```diff
@@ -30,16 +32,28 @@ namespace enigma2
 {
   namespace data
   {
+    class ChannelGroup;
+    typedef std::shared_ptr<enigma2::data::ChannelGroup> ChannelGroupPtr;
```

Comment:
You wanted to avoid inclusion of `Channelgroup.h` which is fine - the price is the copied `typedef`. If you would not use typedefs you couls forward declare enigma2::data::ChannelGroup here and would only need to include `<memory>`

Reply:
Done